src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -102,6 +109,31 @@ internal TElement[] ToArray(int minIdx, int maxIdx)
             return array;
         }
 
+        internal List<TElement> ToList(int minIdx, int maxIdx)
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(source);
+            int count = buffer.count;
+            if (count <= minIdx) return new List<TElement>();
+            if (count <= maxIdx) maxIdx = count - 1;
+            if (minIdx == maxIdx) return new List<TElement>(1) { GetEnumerableSorter().ElementAt(buffer.items, count, minIdx) };
+            int[] map = SortedMap(buffer, minIdx, maxIdx);
+            List<TElement> list = new List<TElement>(maxIdx - minIdx + 1);
+            while (minIdx <= maxIdx)
+            {
+                list.Add(buffer.items[map[minIdx]]);
+                ++minIdx;
+            }
+            return list;
+        }
+
+        internal int GetCount(int minIdx, int maxIdx, bool onlyIfCheap)
+        {
+            int count = GetCount(onlyIfCheap);
+            if (count <= 0) return count;
+            if (count <= minIdx) return 0;
```

Comment:
`count` can. If there's a `-1` signalling a failure to get a cheap path, then this should return `-1`.